bolt/lib/Passes/IdenticalCodeFolding.cpp

```diff
@@ -350,6 +422,9 @@ Error IdenticalCodeFolding::runOnFunctions(BinaryContext &BC) {
   std::atomic<uint64_t> NumFoldedLastIteration{0};
   CongruentBucketsMap CongruentBuckets;
 
+  auto SkipFuncShared = [&](const BinaryFunction &BF) {
```

Comment:
what does `SkipFuncShared` stands for?

Reply:
It's shared by 
ParallelUtilities::PredicateTy SkipFunc and check in createCongruentBuckets.
Not sure what better name is.

Reply:
you can probably just keep both functions and get rid of this lambda

Reply:
Or you can just modify `shouldOptimize()` to check `isSafeToICF()` if it's general enough